configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java

```diff
@@ -158,6 +159,8 @@ public Properties(ApplicationId applicationId,
                     .with(FetchVector.Dimension.APPLICATION_ID, applicationId.serializedForm()).value();
             this.useFdispatchByDefault = Flags.USE_FDISPATCH_BY_DEFAULT.bindTo(flagSource)
                     .with(FetchVector.Dimension.APPLICATION_ID, applicationId.serializedForm()).value();
+            this.useAdaptiveDispatch = Flags.USE_ADAPTIVE_DISPATCH.bindTo(flagSource)
+                    .with(FetchVector.Dimension.ZONE_ID, zone.toString()).value();
```

Comment:
ZONE_ID is not what you think it is. All flags are automatically per zone. The ZONE_ID dimension is only useful if the controller wants to resolve the flag to different values depending on which zone it is operating on.

Consider adding the APPLICATION_ID dimension to allow you to control this per application (and update the Flags definition accordingly).

Reply:
Done, but I hope I can control for the whole zone too......

Reply:
yes